sdk/src/main/java/com/supenta/flitchio/sdk/FlitchioController.java

```diff
@@ -285,161 +280,247 @@ public static int getFlitchioManagerVersionCode(Context context)
+            return;
+        }
 
         boolean willBind = context.bindService(
-                intentToService,
+                new Intent().setClassName(
+                        FLITCHIO_MANAGER_PACKAGE,
+                        FLITCHIO_SERVICE_CLASS),
                 serviceConnection,
                 Context.BIND_AUTO_CREATE);
 
-        activityLifecycleMoment = ActivityLifecycle.ON_CREATE;
-
-        return willBind;
+        if (!willBind) {
+            reportStatus(new Status.FailingStatus(Status.FailingStatus.REASON_SERVICE_UNREACHABLE));
+        } else {
+            reportStatus(new Status(Status.BINDING));
+        }
     }
 
     /**
-     * Register a {@link FlitchioStatusListener} and/or a {@link FlitchioEventListener} to receive
-     * callbacks. If you use this {@link FlitchioController} in an {@link Activity}, this should be
-     * called in your Activity's onResume() (hence the name). If you use this
-     * {@link FlitchioController} in a {@link Service}, this can be called right after
-     * {@link #onCreate()}. You only need to call this if you declare at least one of the two
-     * listeners.
+     * Resume this controller.
+     * <p/>
+     * <strong>You must call this method (or one of its other versions) appropriately in the
+     * lifecycle of your Activity or Service.</strong>
+     * If you use this controller in an {@link Activity}, this method should be called in your
+     * Activity's onResume().
+     * If you use this controller in a {@link Service}, this method can be called right after
+     * {@link #onCreate(FlitchioStatusListener)}.
+     * <p/>
+     * After calling this method, you are ensured to receive at least one
+     * {@link FlitchioStatusListener#onFlitchioStatusChanged(Status)} callback with the current
+     * status.
+     * <p/>
+     * This is a variant of {@link #onResume(FlitchioEventListener)} that allows you to
+     * define the thread on which you wish to receive the event callbacks.
      *
-     * @param statusListener The status listener.
-     * @param eventListener  The event listener.
-     * @param handler        The handler associated to the thread on which the callbacks will
-     *                       happen.
-     * @since 0.6.0
-     */
-    public void onResume(FlitchioStatusListener statusListener, FlitchioEventListener eventListener, Handler handler) {
-        synchronized (lockListener) {
+     * @param eventListener The event listener.
+     * @param handler       The handler associated to the thread on which the callbacks will
+     *                      happen.
+     * @see FlitchioController#onResume()
+     * @see FlitchioController#onResume(FlitchioEventListener)
+     * @since 0.7.0
+     */
+    @MainThread
+    public void onResume(FlitchioEventListener eventListener, Handler handler) {
+        /*
+         * SET UP THE STATUS LISTENER
+         */
+        if (statusListener != null) {
```

Comment:
Nope, in onCreate, the parameter is Nullable. It is explained in the Javadoc: the user still has the possibility to not follow status.